modules/dnn/src/layers/region_layer.cpp

```diff
@@ -91,7 +91,12 @@ class RegionLayerImpl CV_FINAL : public RegionLayer
     {
         CV_Assert(inputs.size() > 0);
         CV_Assert(inputs[0][3] == (1 + coords + classes)*anchors);
-        outputs = std::vector<MatShape>(1, shape(inputs[0][1] * inputs[0][2] * anchors, inputs[0][3] / anchors));
+        int batch_size = inputs[0][0];
+        if(batch_size>1){
+            outputs = std::vector<MatShape>(1, shape(batch_size, inputs[0][1] * inputs[0][2] * anchors, inputs[0][3] / anchors));
+        }else{
+            outputs = std::vector<MatShape>(1, shape(inputs[0][1] * inputs[0][2] * anchors, inputs[0][3] / anchors));
```

Comment:
@kopytjuk, Thank you for your PR. Could you please check what output shape gives Darknet in case of different batch size?

Reply:
Here some example values 
```
batch_size=1: 507 X 85 (n_boxes X (classes+5) )
batch_size=5: 5 X 507 X 85 (batch_size X n_boxes X (classes+5) )
```

Reply:
I did this distinction between different batch sizes to pass all your tests without rewriting them.